src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/process/DefaultProcessRunner.java

```diff
@@ -219,4 +258,72 @@ public void run() {
     return osCommand;
   }
 
+  public static class Builder {
+    private boolean async = false;
+    private ProcessOutputLineListener stdOutLineListener;
+    private ProcessOutputLineListener stdErrLineListener;
+    private ProcessExitListener exitListener;
+    private String waitSuccessMessage;
+    private int waitSuccessTimeoutSeconds = 30;
+
+    /**
+     * Whether to run commands asynchronously.
+     */
+    public Builder async(boolean async) {
+      this.async = async;
+      return this;
+    }
+
+    /**
+     * The client consumer of process standard output.
+     */
+    public Builder stdOutLineListener(ProcessOutputLineListener stdOutLineListener) {
+      this.stdOutLineListener = stdOutLineListener;
+      return this;
+    }
+
+    /**
+     * The client consumer of process error output.
+     */
+    public Builder stdErrLineListener(ProcessOutputLineListener stdErrLineListener) {
+      this.stdErrLineListener = stdErrLineListener;
+      return this;
+    }
+
+    /**
+     * The client listener of the process exit with code.
+     */
+    public Builder exitListener(ProcessExitListener exitListener) {
+      this.exitListener = exitListener;
+      return this;
+    }
+
+    /**
+     * The message to look for in the standard or error output of the process to consider it to be
+     * successfully started. If the message is not seen within the specified timeout {@link
+     * #waitSuccessTimeoutSeconds(int)}, a {@link ProcessRunnerException} will be thrown.
+     */
+    public Builder waitSuccessMessage(String waitSuccessMessage) {
```

Comment:
Yes. It's used when dev app server is started asynchronously: https://github.com/GoogleCloudPlatform/app-maven-plugin/blob/sync_with_lib/src/main/java/com/google/cloud/tools/maven/RunAsyncMojo.java